packages/fuel-gauge/test/fixtures/forc-projects/abi-contract/src/main.sw

```diff
@@ -0,0 +1,1075 @@
+        //     propD2: 40,
+        //     propD3: StructF { propF1: 50, propF2: __to_str_array("ABC") },
+        // };
+        // assert(y == INPUT_Y);
+    }
+
+    fn types_struct_with_implicit_generics(
+        x: StructWithImplicitGenerics<b256, u8>,
+    ) -> StructWithImplicitGenerics<b256, u8> {
+        const INPUT_B256: b256 = 0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa;
+        const INPUT: StructWithImplicitGenerics<b256, u8> = StructWithImplicitGenerics {
+            a: [INPUT_B256, INPUT_B256, INPUT_B256],
+            b: (INPUT_B256, 10),
+        };
+
+        assert(x == INPUT);
+
+        const EXPECTED_B256: b256 = 0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb;
+        const EXPECTED: StructWithImplicitGenerics<b256, u8> = StructWithImplicitGenerics {
+            a: [EXPECTED_B256, EXPECTED_B256, EXPECTED_B256],
+            b: (EXPECTED_B256, 25),
+        };
+
+        EXPECTED
+    }
+
+    fn types_struct_with_array(x: StructWithGenericArray<b256>) -> StructWithGenericArray<b256> {
+        const INPUT_B256: b256 = 0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa;
+        const INPUT_STRUCT: StructDoubleGeneric<b256, u8> = StructDoubleGeneric {
+            a: INPUT_B256,
+            b: 10,
+        };
+        const INPUT: StructWithGenericArray<b256> = StructWithGenericArray {
+            a: [INPUT_STRUCT, INPUT_STRUCT, INPUT_STRUCT],
+        };
+        assert(x == INPUT);
+
+        const EXPECTED_B256: b256 = 0xbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb;
+        const EXPECTED_STRUCT: StructDoubleGeneric<b256, u8> = StructDoubleGeneric {
+            a: EXPECTED_B256,
+            b: 20,
+        };
+        const EXPECTED: StructWithGenericArray<b256> = StructWithGenericArray {
+            a: [EXPECTED_STRUCT, EXPECTED_STRUCT, EXPECTED_STRUCT],
+        };
+
+        EXPECTED
+    }
+
+    fn types_struct_with_vector(x: StructWithVector) -> StructWithVector {
+        let INPUT_VEC: Vec<u8> = vec_u8_from([1, 2, 3]);
+        let INPUT: StructWithVector = StructWithVector {
+            a: 1,
+            b: INPUT_VEC,
+        };
+        assert(x == INPUT);
+
+        let EXPECTED_VEC: Vec<u8> = vec_u8_from([3, 2, 1]);
+        let EXPECTED: StructWithVector = StructWithVector {
+            a: 3,
+            b: EXPECTED_VEC,
+        };
+
+        EXPECTED
+    }
+
+    fn types_struct_with_array_of_enums(x: StructWithEnumArray) -> StructWithEnumArray {
+        const INPUT_ENUM = EnumWithNative::Checked;
+        const INPUT: StructWithEnumArray = StructWithEnumArray {
+            a: [INPUT_ENUM, INPUT_ENUM, INPUT_ENUM],
+        };
+        assert(x == INPUT);
+
+        const EXPECTED_ENUM = EnumWithNative::Pending;
+        const EXPECTED: StructWithEnumArray = StructWithEnumArray {
+            a: [EXPECTED_ENUM, EXPECTED_ENUM, EXPECTED_ENUM],
+        };
+
+        EXPECTED
+    }
```

Comment:
There’s something odd happening here. The `types_struct_with_array_of_enums` function is clearly returning an array of three `EnumWithNative::Pending` values:


```rs
const EXPECTED_ENUM = EnumWithNative::Pending;
const EXPECTED: StructWithEnumArray = StructWithEnumArray {
    a: [EXPECTED_ENUM, EXPECTED_ENUM, EXPECTED_ENUM],
};

EXPECTED
```

Based on this, after decoding, we should see the following result:


```ts
{ a: ['Pending', 'Pending', 'Pending'] }
```

However, when the TS SDK decodes the data, the result is:

```ts
{ a: ['Pending', 'Checked', 'Pending'] }
```

At first, this seems like a decoding issue in the TS SDK. But after inspecting the raw receipt `ReturnData`, I am under the impression that the data returned itself is inconsistent:

```ts
// raw receipt
{
  id: '0xa52541ced4559cf68e923e81ffc900d6855994bfabad4dd9f24c4dfea8e4f183',
  pc: '44544',
  is: '11712',
  ptr: '67107840',
  digest: '0x639809bc6d63798b4571d10a078c77219a7191b65eb30d8b08415d8631c1dbd9',
  len: '24',
  receiptType: 'RETURN_DATA',
  data: '0x000000000000000100000000000000000000000000000001',
}
```

At a first glance, it appears to me that the data should be something homogeneous like:

`0x000000000000000100000000000000010000000000000001`

Since the 3 enum values being returned are exactly the same.

@nedsalk @petertonysmith94 Do you guys know what is happening here?

Reply:
Added the following issue with Sway:
- https://github.com/FuelLabs/sway/issues/6739

Reply:
@petertonysmith94 Nice one. It surely likes something is off within this example.